app/controllers/users_controller.rb

```diff
@@ -1,46 +1,30 @@
 class UsersController < ApplicationController
-  before_action :set_user, only: [:show, :update, :destroy]
+  before_action :set_user, only: [:show]
 
   def index
     @users = User.recent
   end
 
-  def show; end
+  def show
+    @user = User.find_by id: params[:id]
+    return if @user
+
+    flash[:warning] = t("users.not_found")
+    redirect_to root_path
+  end
```

Comment:
```suggestion
  def show; end
```
Code này lặp phải k, a thấy có call set_user trước khi chạy show rồi